src/api/instance/localInstanceImpl.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+
+import {Instance} from "./instance";
+import {type CredrankInput} from "../credrank";
+import {
+  type WeightsT,
+  parser as weightsParser,
+  empty as emptyWeights,
+} from "../../core/weights";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+import stringify from "json-stable-stringify";
+import {
+  type WeightedGraph,
+  fromJSON as weightedGraphFromJSON,
+} from "../../core/weightedGraph";
+import {
+  loadJson,
+  mkdirx,
+  loadFileWithDefault,
+  loadJsonWithDefault,
+} from "../../util/disk";
+import {parser as configParser, type InstanceConfig} from "../instanceConfig";
+import {Ledger} from "../../core/ledger/ledger";
+import {
+  parser as dependenciesParser,
+  type DependenciesConfig,
+} from "../dependenciesConfig";
+import {type Budget} from "../../core/mintBudget";
+import {parser as pluginBudgetParser} from "../pluginBudgetConfig";
+import {
+  CredGraph,
+  parser as credGraphParser,
+} from "../../core/credrank/credGraph";
+
+const DEPENDENCIES_PATH = ["config", "dependencies.json"];
+const WEIGHT_OVERRIDES_PATH = ["config", "weights.json"];
+const BUDGET_PATH = ["config", "pluginBudgets.json"];
+const INSTANCE_CONFIG_PATH = ["sourcecred.json"];
+const CREDGRAPH_PATH = ["output", "credGraph.json"];
+const GRAPHS_DIRECTORY = ["output", "graphs"];
+const GRAPHS_PATH = ["graph.json"];
+const LEDGER_PATH = ["data", "ledger.json"];
+
+/**
+This is an Instance implementation that reads and writes using relative paths
+on the local disk.
+ */
+export class LocalInstance implements Instance {
+  _baseDirectory: string;
+
+  constructor(baseDirectory: string) {
+    this._baseDirectory = baseDirectory;
+  }
+
+  //////////////////////////////
+  //
+  //  Interface Functions
+  //
+  //////////////////////////////
+
+  async loadCredrankInput(): Promise<CredrankInput> {
+    const [
+      pluginGraphs,
+      ledger,
+      weightOverrides,
+      dependencies,
+      pluginsBudget,
+    ] = await Promise.all([
+      await this.loadPluginGraphs(),
+      await this.loadLedger(),
+      await this.loadWeightOverrides(),
+      await this.loadDependencies(),
+      await this.loadPluginsBudget(),
+    ]);
+    return {
+      pluginGraphs,
+      ledger,
+      weightOverrides,
+      dependencies,
+      pluginsBudget,
+    };
+  }
+
+  async loadCredGraph(): Promise<CredGraph> {
+    const credGraphPath = pathJoin(this._baseDirectory, ...CREDGRAPH_PATH);
+    return await loadJson(credGraphPath, credGraphParser);
+  }
+
+  async writeLedger(ledger: Ledger): Promise<void> {
+    const ledgerPath = pathJoin(this._baseDirectory, ...LEDGER_PATH);
+    return fs.writeFile(ledgerPath, ledger.serialize());
+  }
+
+  async writeCredGraph(credGraph: CredGraph): Promise<void> {
+    const cgJson = stringify(credGraph.toJSON());
+    const outputPath = pathJoin(this._baseDirectory, ...CREDGRAPH_PATH);
+    await fs.writeFile(outputPath, cgJson);
```

Comment:
nit: Might as well be consistent about whether we `return` or `await` a
promise in tail position? (I generally like returning them, but at the
behest of others I’ve been convinced to await them instead.)

Reply:
accepted `await`